src/js/components/Button/StyledButtonKind.js

```diff
@@ -32,16 +36,14 @@ const fontStyle = props => {
   `;
 };
 
-const padFromTheme = (size, theme) => {
-  if (
-    size &&
-    theme.button.size &&
-    theme.button.size[size] &&
-    theme.button.size[size].pad
-  ) {
+const padFromTheme = (size, theme, kind) => {
+  // caller has specified a themeObj to use for styling
+  // relevant for cases like pagination which looks to theme.pagination.button
+  const themeObj = typeof kind === 'object' ? kind : theme.button;
```

Comment:
We seem to be doing this check in several areas. I wonder if we can further simplify by performing the check once on L109 and then modifying the function calls to be a bit 'dumber' and always use whatever kind is passed to them.

Line 109 would read `const themeObj = typeof kind === 'object' ? kind : theme.button;`

Reply:
It's a bit difficult to keep track of which theme object I should be using in each scenario, whether I should be setting it to theme.button vs. undefined. If we could make that decision once, then call the functions with the appropriate values..

Reply:
Good call, I'll look into this cleanup